src/System.Collections.Specialized/tests/HybridDictionary/SetItemObjObjTests.cs

```diff
@@ -647,5 +644,17 @@ public void Test01()
                 Assert.False(true, string.Format("Error, failed to set to null"));
             }
         }
+
+        [Fact]
+        public void Set_CaseInsensitive_Works()
```

Comment:
Not for setting a key on a case insensitive hybrid dictionary where the key to set is the same case as the key added.